File: convert.ts
import * as ecc from '@bitcoin-js/tiny-secp256k1-asmjs';
import * as bitcoin from 'bitcoinjs-lib';
import { ECPairFactory, ECPairInterface } from 'ecpair';
import { Command } from 'commander';
import { Options } from './types';

const ECPair = ECPairFactory(ecc);
const program = new Command();
const opts = program.opts<Options>();

const hexToWIF = (hexKey: string, testnet = false, compressed = true) => {
  if (!hexKey) throw new Error('Missing private key.');
  if (!/^[0-9a-fA-F]{64}$/.test(hexKey)) throw new Error('Invalid private key. Must be a 64-character hex string.');
  const network = testnet ? bitcoin.networks.testnet : bitcoin.networks.bitcoin;
  const pair: ECPairInterface = ECPair.fromPrivateKey(Buffer.from(hexKey, 'hex'), { network, compressed });
  return pair.toWIF();
};

program
  .name('convert')
  .description('Convert Bitcoin private key (hex) to WIF format')
  .requiredOption('-k, --key <hex>', 'Private key in hex format')
  .option('--testnet', 'Use Bitcoin testnet network')
  .option('-u', '--uncompressed', 'Generate uncompressed WIF key')
  .parse(process.argv);

try {
  const wif = hexToWIF(opts.key, opts.testnet, !opts.uncompressed);
  console.log(`WIF (${opts.testnet ? 'testnet' : 'mainnet'}, ${opts.uncompressed ? 'uncompressed' : 'compressed'}): ${wif}`);
} catch (err) {
  console.log((err as Error).message);
  process.exit(1);
}

File: src/convert.rs
use bitcoin::{Address, Network, NetworkKind, PrivateKey, PublicKey};

pub fn run_convert(hex_key: &str, testnet: bool, uncompressed: bool) -> Result<String, Box<dyn std::error::Error>> {
    let key_bytes = hex::decode(hex_key.trim_start_matches("0x")).map_err(|_| "Invalid hex key format")?;
    if key_bytes.len() != 32 {
        return Err("Private key must be 32 bytes".into());
    }
    let network = if testnet { Network::Testnet } else { Network::Bitcoin };
    let network_kind: NetworkKind = network.into();
    let secp = bitcoin::secp256k1::Secp256k1::new();
    let sk = bitcoin::secp256k1::SecretKey::from_slice(&key_bytes)?;
    let private_key = PrivateKey { compressed: !uncompressed, network: network_kind, inner: sk };
    let pk = PublicKey::from_private_key(&secp, &private_key);
    let address = Address::p2pkh(pk, network);
    let wif = private_key.to_wif();
    let output = format!(
        "Network: {}\n\
         Format: {}\n\
         WIF: {}\n\
         Address: {}",
        if testnet { "Testnet" } else { "Mainnet" },
        if uncompressed { "Uncompressed" } else { "Compressed" },
        wif,
        address
    );
    Ok(output)
}
